processing/src/main/java/io/druid/segment/filter/ExtractionFilter.java

```diff
@@ -54,15 +56,20 @@ public ExtractionFilter(
   {
     final Indexed<String> allDimVals = selector.getDimensionValues(dimension);
     final List<Filter> filters = Lists.newArrayList();
-    if (allDimVals != null) {
-      for (int i = 0; i < allDimVals.size(); i++) {
+    if (allDimVals != null)
+    {
+      for (int i = 0; i < allDimVals.size(); i++)
+      {
         String dimVal = allDimVals.get(i);
-        if (value.equals(fn.apply(dimVal))) {
+        if (value.equals(fn.apply(dimVal)))
```

Comment:
as you can see this is old druid code (that i am using to understand how it works) but i guess you are right.
so you are saying that if the lookup map is 
```
{
"dim1":null
"dim2":""
}
```
apply on both will always return `null` while filter value can be only empty string `""` ?